File: ios/Runner/CookieExtractorPlugin.swift
import Flutter
import WebKit

class CookieExtractorPlugin: NSObject, FlutterPlugin {
    static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(
            name: "frosty/cookie_extractor",
            binaryMessenger: registrar.messenger()
        )
        let instance = CookieExtractorPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }

    func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        if call.method == "extractTwitchAuthToken" {
            WKWebsiteDataStore.default().httpCookieStore.getAllCookies { cookies in
                let authCookie = cookies.first {
                    $0.name == "auth-token" && $0.domain.contains("twitch.tv")
                }
                result(authCookie?.value)
            }
        } else {
            result(FlutterMethodNotImplemented)
        }
    }
}

File: android/app/src/main/kotlin/com/tommychow/frosty/CookieExtractorPlugin.kt
package com.tommychow.frosty

import android.webkit.CookieManager
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class CookieExtractorPlugin : FlutterPlugin, MethodChannel.MethodCallHandler {
    private lateinit var channel: MethodChannel

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(binding.binaryMessenger, "frosty/cookie_extractor")
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        if (call.method == "extractTwitchAuthToken") {
            val cookies = CookieManager.getInstance().getCookie("https://twitch.tv")
            val authToken = cookies?.split(";")
                ?.map { it.trim() }
                ?.firstOrNull { it.startsWith("auth-token=") }
                ?.substringAfter("auth-token=")
            result.success(authToken)
        } else {
            result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}
